[export] Allow None as the meta value for tensor output.

zhxchen17 opened the pull request:
Summary: Sometimes we will get a None value from ops which returns Tensor type in the schema. Allow this case during serialization.

Test Plan: test__scaled_dot_product_flash_attention

Differential Revision: D52491668

SherlockNoMad commented:
As discussed offline, this is an unfortunate exception we have to make, as schema for `_scaled_dot_product_flash_attention` schema is incorrect. In some cases, it returns None for `Tensor`-typed output.

zhxchen17 committed:
[export] Allow None as the meta value for tensor output. (#116664)

Summary:

Sometimes we will get a None value from ops which returns Tensor type in the schema. Allow this case during serialization.

Test Plan: test__scaled_dot_product_flash_attention

Reviewed By: SherlockNoMad

Differential Revision: D52491668
